### PR TITLE
ENG-11657 do not resume replication for DR consumer if durability is off

### PR DESCRIPTION
**Pull request by dhan111**
If you use the new CLI without command logging, the database will pick the recover start action internally when you "start" after the first time even though there is nothing to recover from. This tricks the DR system into thinking that it can resume replication. It eventually fails of course, but it may confuse the remote cluster and generate confusing error messages.

Now before resuming replication, we check if durability is turned on and if there is a completed snapshot. If durability is off or no snapshot is found, we start fresh.

### CODE REVIEW

File: src/frontend/org/voltdb/RealVoltDB.java [hunk 1]
```diff
@@ -1039,14 +1041,15 @@ public void initialize(Configuration config) {
             validateStartAction();
 
             // durable means commandlogging is enabled.
-            boolean durable = readDeploymentAndCreateStarterCatalogContext(config);
+            m_durable = readDeploymentAndCreateStarterCatalogContext(config);
+
```
Comment (JadenW) [inline src/frontend/org/voltdb/RealVoltDB.java]: So m_durable is only set in the initialize time and don't need to be updated when commandlogging setting changes (i.e.  change commandlogging from disable to enable via catalog update)?  
I think this is fine as m_durable is used for determining resumeReplication of consumer gateway (which happens before any catalog update).

Comment (JadenW) [inline src/frontend/org/voltdb/RealVoltDB.java]: nitpick: change durable to m_durable in the comment and maybe move this comment to its definition.

Comment (manjujames) [inline src/frontend/org/voltdb/RealVoltDB.java]: I don't think we allow enabling command logging after server is up.

**Comment by manjujames** [comment]
It looks good to me. We will look at jenkins today before merging.

File: src/frontend/org/voltdb/RealVoltDB.java [hunk 2]
```diff
@@ -1571,6 +1574,18 @@ public Boolean get() {
     }
 
     /**
+     * Check if replication needs to be resumed
+     * Return true replication needs to be resumed,
+     * false if we need to starting new or command log was disabled or
+     * there was no complete snapshot
+     */
+   private boolean willResumeReplication()
```
Comment (slowsoul) [inline src/frontend/org/voltdb/RealVoltDB.java]: Nitpicking: renaming this method to willDoActualRecover() may be more accurate.

**Pull request merged**